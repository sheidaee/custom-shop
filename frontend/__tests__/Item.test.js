import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: '123456',
  title: 'The title',
  price: 100,
  description: 'The description',
  image: 'item.jpg',
  largeImage: 'large-image.jpg'
}

describe('<Item />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<ItemComponent item={fakeItem} />);
  })

  it('renders and matches the snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
  
  })
})