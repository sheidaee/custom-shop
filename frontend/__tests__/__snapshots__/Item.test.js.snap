// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="The title"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123456",
          },
        }
      }
    >
      <a>
        The title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $1
  </PriceTag>
  <p>
    The description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123456",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="123456"
    />
    <DeleteItem
      id="123456"
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
